src/EnergyPlus/HVACInterfaceManager.cc

```diff
@@ -128,12 +127,13 @@ void UpdateHVACInterface(EnergyPlusData &state,
             totDemandSideMaxAvail += state.dataLoopNodes->Node(demInNode).MassFlowRateMaxAvail;
         }
         TmpRealARR = state.dataConvergeParams->AirLoopConvergence(AirLoopNum).HVACFlowDemandToSupplyTolValue;
-        state.dataConvergeParams->AirLoopConvergence(AirLoopNum).HVACFlowDemandToSupplyTolValue(1) =
+        state.dataConvergeParams->AirLoopConvergence(AirLoopNum).HVACFlowDemandToSupplyTolValue[0] =
             std::abs(totDemandSideMassFlow - state.dataLoopNodes->Node(InletNode).MassFlowRate);
-        state.dataConvergeParams->AirLoopConvergence(AirLoopNum).HVACFlowDemandToSupplyTolValue({2, DataConvergParams::ConvergLogStackDepth}) =
-            TmpRealARR({1, DataConvergParams::ConvergLogStackDepth - 1});
-        if (state.dataConvergeParams->AirLoopConvergence(AirLoopNum).HVACFlowDemandToSupplyTolValue(1) > DataConvergParams::HVACFlowRateToler) {
-            state.dataConvergeParams->AirLoopConvergence(AirLoopNum).HVACMassFlowNotConverged(1) = true;
+        for (int logIndex = 2; logIndex <= DataConvergParams::ConvergLogStackDepth; logIndex++) {
```

Comment:
Eliminated the index-range argument based shifting being done here and used a simple loop.  I'm open to suggestion, but this is exactly what the code was doing inside the Array1D library anyway, so there shouldn't be a performance issue, and this just makes it look much more clear to anyone looking at this code in the future.